xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/ViewFile.xml

```diff
@@ -263,24 +404,161 @@
+            $escapetool.xml($services.localization.render('cancel'))&lt;/button&gt;
+        &lt;/div&gt;
+      &lt;/div&gt;
+    &lt;/div&gt;
+  &lt;/div&gt;
+#end
 #macro (executeMacro)
+  #set ($discard = $xwiki.ssx.use('Confluence.Macros.ViewFile'))
   #set($hasPDFViewer = $xwiki.exists("XWiki.PDFViewerMacro"))
   #set($officeExtensions = [ 'ppt', 'pptx', 'odp', 'doc', 'docx', 'odt', 'xls', 'xlsx', 'ods' ])
+  #set($presentationExtensions = [ 'ppt', 'pptx', 'odp' ])
+  #set($spreadsheetExtensions = [ 'xls', 'xlsx', 'ods' ])
   #set($unescapedFilename = $xcontext.macro.params.get('att--filename'))
   #if(!$unescapedFilename)
     #set($unescapedFilename = $xcontext.macro.params.get('name'))
   #end
   #set($extension = $unescapedFilename.substring($mathtool.add($unescapedFilename.lastIndexOf('.'), 1)).toLowerCase())
   #set($escapedFilename = $services.rendering.escape($unescapedFilename, $xwiki.currentContentSyntaxId))
-  #if($extension == 'pdf' &amp;&amp; $hasPDFViewer)
+  #set($width = $xcontext.macro.params.get('width'))
+  #set($height = $xcontext.macro.params.get('height'))
+  #set($display = $xcontext.macro.params.get('display'))
+  {{html wiki="true" clean="false"}}
+    #if ($display == 'FULL')
+      #if ($presentationExtensions.contains($extension))
+        #if($height &gt; 390)
+          #set($height = '390')
+        #end
 
-    {{pdfviewer file="$escapedFilename" /}}
-  #elseif($officeExtensions.contains($extension))
+        #customPresentationBox($width $height $escapedFilename)
+      #elseif($extension == 'pdf' &amp;&amp; $hasPDFViewer)
 
-    {{office reference="attach:$escapedFilename" /}}
-  #elseif($doc.getAttachment($unescapedFilename))
-    [[attach:$escapedFilename]]
-  #end
+        {{pdfviewer file="$escapedFilename" height=$height width=$width/}}
+      #elseif($officeExtensions.contains($extension))
+
+        #customBox($width $height $escapedFilename)
+      #elseif($doc.getAttachment($unescapedFilename))
+        #set ($downloadUrl = $xwiki.getAttachmentURL($doc.getFullName(), $unescapedFilename))
+        &lt;div&gt;
+          &lt;a href="$downloadUrl" download&gt;$escapedFilename&lt;br&gt;&lt;/a&gt;
+        &lt;/div&gt;
+      #end
+    #elseif ($display == 'THUMBNAIL')
+      #if (($extension == 'pdf' &amp;&amp; $hasPDFViewer) || $officeExtensions.contains($extension))
+        #set ($modalId = $escapetool.xml($unescapedFilename.replaceAll("[^a-zA-Z]", "") + 'DisplayContentModal'))
+        #if ($presentationExtensions.contains($extension))
+          &lt;div class="buttonwrapper"&gt;
+            &lt;a href="#${modalId}" data-toggle="modal" data-target="#${modalId}"&gt;
+              $services.icon.render('file-powerpoint') $escapedFilename&lt;/a&gt;
+          &lt;/div&gt;
+          #displayPresentationContent($escapedFilename, $modalId)
+        #else
+          #if($extension == 'pdf')
+            &lt;div class="buttonwrapper"&gt;
```

Comment:
You should also have a "title" attribute for the buttons that would say something along the lines "Click to view the file". To indicate further that the button is a thumbnail.

You shouldnt have an `a` tag inside the button. I reckon you placed it like this to have the `onhover` effect. It will probably trigger only when hovering the text inside the button. You should add a css rule for the button so that the text will get underlined even if you hover on the wrapper of the text.

You can also have a velocity macro for rendering the button, since you duplicate the code.